src/arrayHandling.ts

```diff
@@ -64,6 +65,15 @@ export function handleArrayLogic<Variables>(
                     return item
                 }
 
+                // handle properties of an object
+                if (typeof item === "object") {
+                    const newReference = `$${reference.substring(
+                        reference.indexOf("#.") + 1,
```

Comment:
I thought it might, but since it's actually checking if it's found, and the logic before it returns something if found. There probably needs to be a test for this.